packages/cli/src/commands/screenshot-diff/screenshot-diff.command.ts

```diff
@@ -23,79 +26,198 @@ export class DiffError extends Error {
       baseReplayId: string;
       headReplayId: string;
       threshold: number;
-      value: number;
+      value?: number;
     }
   ) {
     super(message);
   }
 }
 
+type ComparisonOutcome = "pass" | "fail";
+
+export interface ScreenshotDiffResult {
+  baseScreenshotFile: string;
+  headScreenshotFile: string;
+
+  outcome: ComparisonOutcome;
+  comparisonResult: CompareImageResult;
+}
+
 export const diffScreenshots: (options: {
   client: AxiosInstance;
   baseReplayId: string;
   headReplayId: string;
-  baseScreenshot: PNG;
-  headScreenshot: PNG;
+  baseScreenshotsDir: string;
+  headScreenshotsDir: string;
   threshold: number | null | undefined;
   pixelThreshold: number | null | undefined;
   exitOnMismatch: boolean;
-}) => Promise<void> = async ({
+}) => Promise<ScreenshotDiffResult[]> = async ({
   client,
   baseReplayId,
   headReplayId,
-  baseScreenshot,
-  headScreenshot,
+  baseScreenshotsDir,
+  headScreenshotsDir,
   threshold: threshold_,
   pixelThreshold,
   exitOnMismatch,
 }) => {
   const logger = log.getLogger(METICULOUS_LOGGER_NAME);
 
   const threshold = threshold_ || DEFAULT_MISMATCH_THRESHOLD;
```

Comment:
I think this is a mistake from me:

```suggestion
  const threshold = threshold_ ?? DEFAULT_MISMATCH_THRESHOLD;
```